R/save.r

```diff
@@ -149,8 +149,12 @@ plot_dev <- function(device, filename = NULL, dpi = 300) {
   force(filename)
   force(dpi)
 
-  if (is.function(device))
-    return(device)
+  if (is.function(device)) {
+    if ("file" %in% names(formals(device))) {
+      dev <- function(filename, ...) device(file = filename, ...)
+      return(dev)
+    } else return(device)
```

Comment:
According to https://style.tidyverse.org/syntax.html#inline-statements:

> Function calls that affect control flow (like `return()`, `stop()` or continue) should always go in their own `{}` block:

```suggestion
    } else {
      return(device)
    }
```